uat/testing-features/src/main/java/com/aws/greengrass/LocalDeploymentSteps.java

```diff
@@ -0,0 +1,266 @@
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+import javax.inject.Inject;
+
+import static com.aws.greengrass.testing.component.LocalComponentPreparationService.ARTIFACTS_DIR;
+import static com.aws.greengrass.testing.component.LocalComponentPreparationService.LOCAL_STORE;
+import static com.aws.greengrass.testing.component.LocalComponentPreparationService.RECIPE_DIR;
+import static com.aws.greengrass.testing.features.GreengrassCliSteps.LOCAL_DEPLOYMENT_ID;
+
+@ScenarioScoped
+public class LocalDeploymentSteps {
+    private static final Logger LOGGER = LogManager.getLogger(LocalDeploymentSteps.class);
+    private static final String MERGE_CONFIG = "MERGE";
+    private static final Path LOCAL_STORE_RECIPES = Paths.get("local:", "local-store", "recipes");
+    private static final int MAX_DEPLOYMENT_RETRY_COUNT = 3;
+    private final ComponentPreparationService componentPreparation;
+    private final ComponentOverrides overrides;
+    private final TestContext testContext;
+    private final WaitSteps waits;
+    private final ObjectMapper mapper;
+    private final ScenarioContext scenarioContext;
+    private final Path artifactPath;
+    private final Path recipePath;
+    private final Device device;
+
+    @Inject
+    @SuppressWarnings("MissingJavadocMethod")
+    public LocalDeploymentSteps(
+            final ComponentOverrides overrides,
+            final TestContext testContext,
+            final ComponentPreparationService componentPreparation,
+            final ScenarioContext scenarioContext,
+            final WaitSteps waits,
+            final ObjectMapper mapper,
+            final Device device) {
+        this.overrides = overrides;
+        this.testContext = testContext;
+        this.componentPreparation = componentPreparation;
+        this.scenarioContext = scenarioContext;
+        this.waits = waits;
+        this.mapper = mapper;
+        this.artifactPath = testContext.installRoot().resolve(LOCAL_STORE).resolve(ARTIFACTS_DIR);
+        this.recipePath = testContext.installRoot().resolve(LOCAL_STORE).resolve(RECIPE_DIR);
+        this.device = device;
+    }
+
+    /**
+     * implemented the step of installing a custom component with configuration.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I install the component {word} from local store with configuration")
```

Comment:
why is this here?

Local deployments should be implemented in OTF itself, this is not specific to Log Manager. Can we do this better?

Reply:
OTF has local deployment but it does not take parameters. The LogManager has it here so then it can take some parameters for different scenarios, e.g. logging frequency, file size, rotation speed.

Reply:
OTF should have this logic.

Reply:
But for different component, it is hard to define the number of parameters. Will it be more component-owned thing, instead of general steps in OTF?  Here is the [link](https://github.com/aws-greengrass/aws-greengrass-testing/blob/1ce7ef461465fb11f701f2e1231caa817afcba04/aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java#L148) for what original OTF has:

Reply:
This will be moved into OTF eventually. Now, we are trying what we want for local deployments and will provide feedback to OTF team.

Reply:
We're all the same team. Why do you add the appropriate code into OTF and then consume it here? We don't need to wait for them to do it, you've already done it; just add it into OTF with a pull request.

Reply:
The codes have been merged into OTF. The new commit is letting LogManager directly consume methods from OTF.